h_malloc.c

```diff
@@ -1874,6 +1874,94 @@ EXPORT size_t h_malloc_object_size_fast(const void *p) {
     return SIZE_MAX;
 }
 
+inline void *h_memcpy_real(void *dst, const void *src, size_t len) {
+#if CONFIG_BLOCK_OPS_CHECK_SIZE && !defined(HAS_ARM_MTE)
+    char *p_dst = (char *)dst;
+    char const *p_src = (char const *)src;
+
+    while(len--) {
+        *p_dst++ = *p_src++;
+    }
+
+    return dst;
+#else
+    return memcpy(dst, src, len);
+#endif
+}
+
+inline void *h_memmove_real(void *dst, const void *src, size_t len) {
+#if CONFIG_BLOCK_OPS_CHECK_SIZE && !defined(HAS_ARM_MTE)
```

Comment:
I think it would make sense to make `CONFIG_BLOCK_OPS_CHECK_SIZE` incompatible with `HAS_ARM_MTE` at compilation time, instead of silently ignoring the former in the presence of the latter.

Reply:
I know how to do that for the Makefile but am not sure how to do that for the Android.bp

Reply:
You can use a `static_assert`, like [here](https://github.com/GrapheneOS/hardened_malloc/blob/main/h_malloc.c#L31)